HelloQuad/HelloQuad.cs

```diff
@@ -0,0 +1,191 @@
+
+        texture = new Texture2D();
+        texture.SetData(image);
+
+        // We create a new VertexBuffer object, it holds our vertices and is passed to the GPU
+        vertexBuffer = new VertexBuffer();
+        // We set its size and the elements it's containing, the 3rd optional argument (dynamic) should be 'true' if you're planning
+        // to update the VertexBuffer constantly, that will improve performance in those cases.
+        vertexBuffer.SetSize(6, Constants.MASK_POSITION | Constants.MASK_TEXCOORD1, false);
+
+        // Here we lock the vertexBuffer what returns a pointer (IntPtr) to its data (vertexData here), I'm using a code block for clarity
+        System.IntPtr vertexData = vertexBuffer.Lock(0, 6, true);
+        {
+            // We can cast the data pointer to whatever data type we want, here we are only using floats but ideally you will want
+            // to cast it to an object (struct) with properly offsetted fields and maybe unions for things like colors
+            float* vout = (float*)vertexData;
+
+            // Our first vertex, here we set the x position of it
+            *vout++ = 0;
+            // Here we set the y position
+            *vout++ = 0;
+            // Here we set the z position (depth in this case, useful for sorting in orthographic projection)
+            *vout++ = 0;
+            // Here we set it's texture x coordinate, commonly called u;
+            *vout++ = 0;
+            // Here we set it's texture y coordinate, commonly called v;
+            // UVs are simply cartesian coordinates: 0,0 is bottom-left; 1,1 is top-right
+            *vout++ = 0;
+
+            // Each of these blocks is a vertex, same concept apply:
+            *vout++ = 0; // x
+            *vout++ = 1; // y
+            *vout++ = 0; // z
+            *vout++ = 0; // u
+            *vout++ = 1; // v
+
+            *vout++ = 1;
+            *vout++ = 1;
+            *vout++ = 0;
+            *vout++ = 1;
+            *vout++ = 1;
+
+            *vout++ = 0;
+            *vout++ = 0;
+            *vout++ = 0;
+            *vout++ = 0;
+            *vout++ = 0;
+
+            *vout++ = 1;
+            *vout++ = 1;
+            *vout++ = 0;
+            *vout++ = 1;
+            *vout++ = 1;
+
+            *vout++ = 1;
+            *vout++ = 0;
+            *vout++ = 0;
+            *vout++ = 1;
+            *vout++ = 0;
+
+        }
+        // Don't forget to unlock the VertexBuffer after you modify it
+        vertexBuffer.Unlock();
+
+    }
+
+    void Render()
+    {
+        // We clear the whole screen white before drawing anything
+        graphics.Clear(0x1, Color.White);
+        // The 3 lines below don't have to be set every frame in this specific example, but you'll most likely be changing the often
+        viewport.View.SetCameraShaderParameters(camera);
+        // We set the Texture to be used in the next draw call
+        graphics.SetTexture((uint)TextureUnit.TU_DIFFUSE, texture);
```

Comment:
Yes, casting definitely doesn't look good. I don't think it's possible to had implicit enum conversion, but both the 1st and 2nd options look OK to me, however, I don't know what exactly would be the implications of either. If there are no major cons I'd say 1st is better.